### ISSUE TITLE
Offer/suggest Mimic as a TTS reader in the preferences?

### ISSUE THREAD

**Issue by nekohayo** [johnfactotum/foliate#618]
https://mimic.mycroft.ai, if present on the system, could be offered as a preset choice to the user in the preferences to make things easier (and to let people discover the existence of Mimic, perhaps). So far, the output quality seems to be vastly superior to espeak (I don't know about festival, but I suspect it's better than festival too). For what it's worth, mimic is available as a package in Fedora.

**Comment by johnfactotum** [comment]
Thanks for the suggestion. Didn't know about Mimic. Looks very interesting.

For what it's worth, festival actually produces very decent results (much, much better than espeak, anyway), but it's also quite slow, and the range of supported languages is more limited.

**Comment by itprojects** [comment]
@nekohayo @johnfactotum The following instructions should work for Fedora 33, when mimic is already installed.

1. Open Foliate
2. Go Preferences
3. Press "Configure" in the TTS section.
4. Choose "Other"
5. Put in the empty box the following command:

> ./mimic -f /dev/stdin

or with a different voice

> mimic -f /dev/stdin -voice slt

Other available voices are: ap slt slt_hts kal awb kal16 rms awb_time

![1](https://user-images.githubusercontent.com/2698054/108002623-175d0500-6fe8-11eb-983d-cf473f1e73f3.png)

**Comment by itprojects** [comment]
@johnfactotum Answered; close?

**Comment by nekohayo** [comment]
I know you can manually put it there as a custom command; but my point was about discoverability and convenience, the same way as (from my understanding) espeak and festival work. I still think it should be a simple radiobutton for mimic when available on the system (and the radiobuttons for espeak, festival, mimic, etc. should be individually marked as (in)sensitive depending on the installed availability of those libraries/binaries on the system). In other words, Mimic should be given equal footing to the other options (if not priority, because it yields muuuuch better results).

**Comment by itprojects** [comment]
@nekohayo Questions for discussion:

(Apologies if these are newbie, too simple.)

1. About the radio button, different systems may require different detection, which is so much more messy that a simple box, in which the user places the configuration string.

2. Do you know how to install/configure mimic on Debian-based systems?

3. Does the user have to have an account to use newer `mimic` versions?

4. Does mimic transmit the text strings over the network? Can it, for better quality?

About the equal footing:

>  Mimic should be given equal footing to the other options (if not priority, because it yields muuuuch better results).

Yes, the quality is much better. The closest rival is Mbrola voices, even if such judgement is subjective.  Running Tacotron2 locally, for Foliate would be practically human speech (after some exponential improvement in computation speeds).

**Comment by apandada1** [comment]
@itprojects I found some instructions for compiling mimic in Debian based systems (but I have not tried compiling it). https://github.com/MycroftAI/mimic1#linux

**Comment by itprojects** [comment]
One major problem with Mimic is that it doesn't seem to be directly installable from a suported package (apt-get). The build instructions are not something a casual user would easily understand, so it may be too early for default adoption into Foliate.

**Comment by unomi-zan** [comment]
So, apparently `mimic3` has been released and it sounds so much better than `festival` and `espeak`.

And it can be easily be installed from deb packages and pip. This would be a great addition to foliate.

**Comment by johnfactotum** [comment]
Currently the plan is to switch to speech-dispatcher (not done yet, but there's already an SSIP client implementation in the `gtk4` branch). And it seems Mimic 3 has some sort of work-in-progress support for speech-dispatcher as well.

**Comment by nekohayo** [comment]
> One major problem with Mimic is that it doesn't seem to be directly installable from a suported package (apt-get). 

For what it's worth, on Fedora mimic is directly installable from the main repositories.

It could be a "soft" dependency of foliate if you wish to push its adoption across distros.

**Comment by securegh** [comment]
> So, apparently `mimic3` has been released and it sounds so much better than `festival` and `espeak`.

Hello @unomi-zan , 

Could you please show how can I setup mimic3 with Foliate in the preference menu?  I have been struggling to make mimic3 work with Foliate. 

Thanks a lot!

**Comment by unomi-zan** [comment]
@securegh I'm sorry but i also don't know what that error is. I don't have programming knowledge

**Comment by securegh** [comment]
> @securegh I'm sorry but i also don't know what that error is. I don't have programming knowledge

Have you been able to setup mimic3 on Foliate?

**Comment by unomi-zan** [comment]
@securegh no. I have given up on that. I can't even use mimic3 in fedora now because of the python version issue.
